simulator/include/simulator.hpp

```diff
@@ -338,7 +344,9 @@ class TestRunner {
         auto instances = get_instances();
 
         for (int i = 0; i < instances; i++) {
-            schedule_producer(now + i * get_slot_ms(), ordering[i]);
+            if(nodetypes.at(i) != node_type::FN) {
```

Comment:
I think it will be more reasonable to add an `assert(nodetypes.size() == delay_matrix.size());` somewhere in the beginning of the `run()`, and use `operator[]` for `nodetypes`.

Reply:
fixed in next commit